### DISCUSSION TITLE
Passing in javascript function in index.html to generate timeline variables

### DISCUSSION THREAD

**Discussion by synapticlee** [jspsych/jsPsych#2941] [Q&A]
Hi, I defined a javascript [function/script] to define an object for timeline_variables with [stimulus: blah.png, {data: {correct_response: blah}}, etc etc]. However I'm having trouble understanding how I can call this function within index.html. I have loaded the script in the html at the top with <script ..></script> but when I try to call the function e.g. foo = my_function(), it doesn't work, nor is just calling my_script() directly. How do I pass the output of this function to index.html so that I can use it in timeline_variables: [output]?

EDIT: never mind, the bug was due to a different error. for now I'm doing 
var output = my_function() and timeline_variables = output, and it seems to work. Would still like to know what is the best practice/syntax for this type of operation.

**Comment by jodeleeuw** [comment]
> Would still like to know what is the best practice/syntax for this type of operation.

I think the way that you've described it seems perfectly reasonable. Just to recap for the sake of this thread:

You can have a file like `generate_timeline.js` that contains code like this:

```js
function createTimelineVariables(){
   // do stuff here...
  return timeline_variable_array;
}
```

In `index.html` you can load this script:

```html
<script src="generate_timeline.js"></script>
```

And also call the function:

```js
<script>
  var timeline_variables = createTimelineVariables();
</script>
```
